content/automate/ManageIQ/AutomationManagement/AnsibleTower/Operations/Methods.class/__methods__/available_credentials.rb

```diff
@@ -63,6 +63,4 @@ def fill_dialog_field(list)
   end
 end
 
-if __FILE__ == $PROGRAM_NAME
-  ManageIQ::Automate::AutomationManagement::AnsibleTower::Operations::AvailableCredentials.new.main
-end
+ManageIQ::Automate::AutomationManagement::AnsibleTower::Operations::AvailableCredentials.new.main
```

Comment:
Is that the intention?  To instantiate this class unconditionally?

Reply:
Yes.  In test, https://github.com/ManageIQ/manageiq-content/pull/390/files#diff-93830fa29d616f7c87903d08b5b1b29aR66 will catch this on load and raise if the constant doesn't exist.  I don't think there is any other use case for this conditional.

Reply:
What about non-test code?   Does this file get required ever?

Reply:
@mkanoor can you confirm that this is never `require`d or `load`ed?

Reply:
@bdunne That if conditional is to only allow the method to be called when its executed from the Automate engine. If we remove the condition it breaks that logic.

Reply:
@mkanoor The two cases I know of are:
- Tests: we don't want to run the code
- Execution: we want the code to run in the automation engine

Are there any other cases where these files are required or loaded and we don't want it to run?

Reply:
@bdunne 
During testing we only want to run the code when we have a test setup done, so that the invocation doesn't fail. That would give us the 100% coverage. Today we test everything for these new methods except for the if block code.

Not all newer classes have the If Block.